src/psbt.h

```diff
@@ -36,6 +36,14 @@ static constexpr uint8_t PSBT_OUT_BIP32_DERIVATION = 0x02;
 // as a 0 length key which indicates that this is the separator. The separator has no value.
 static constexpr uint8_t PSBT_SEPARATOR = 0x00;
 
+/** An exception in PSBT processing that represents a problem with the input data --
+    when the input data is user-supplied, it is appropriate to catch() this and use
+    e.what() to retrieve a human-readable error message. */
+class PSBTException: public std::runtime_error {
```

Comment:
This results in RPC errors with a code `RPC_MISC_ERROR`. I think it would be better if you could somehow make the error code more specific (the message still comes through).

Reply:
Yeah, I'd love any thoughts on how to accomplish this without gunking up non-RPC code too much when it calls this codepath. I don't really want this to depend on the RPC stuff.